File: data/readable_format/fortran/file914.f90
subroutine sedivm(n, dx, incx, dy, incy)
  implicit none
  integer, intent(in) :: n, incx, incy
  real, intent(in) :: dx(*)
  real, intent(inout) :: dy(*)
  integer :: i, ix, iy

  if (n.le.0) return
  ix = 1
  iy = 1
  if (incx < 0) ix = (-n + 1) * incx + 1
  if (incy < 0) iy = (-n + 1) * incy + 1

  do i = 1, n
     dy(iy) = dy(iy) / dx(ix)
     ix = ix + incx
     iy = iy + incy
  end do
end subroutine sedivm

program test
  implicit none
  real, allocatable :: dx(:), dy(:)
  integer :: n, incx, incy

  ! Test 1
  n = 5
  allocate(dx(n), dy(n))
  dx = (/1, 2, 3, 4, 5/)
  dy = (/10, 20, 30, 40, 50/)
  incx = 1
  incy = 1
  call sedivm(n, dx, incx, dy, incy)
  print *, 'Test 1:', dy

  ! Test 2
  dx = (/1, 2, 3, 4, 5/)
  dy = (/5, 10, 15, 20, 25/)
  incx = 2
  incy = 2
  call sedivm(3, dx, incx, dy, incy)
  print *, 'Test 2:', dy

  ! Test 3
  dx = (/5, 4, 3, 2, 1/)
  dy = (/50, 40, 30, 20, 10/)
  incx = -1
  incy = -1
  call sedivm(5, dx, incx, dy, incy)
  print *, 'Test 3:', dy

  deallocate(dx, dy)
end program test

File: data/readable_format/cpp/file914.cpp
#include <iostream>
#include <vector>
#include <cassert>
#include <cmath>

void sedivm(int n, float* dx, int incx, float* dy, int incy) {
    if (n <= 0) return;

    int ix = (incx > 0) ? 0 : (-n + 1) * incx;
    int iy = (incy > 0) ? 0 : (-n + 1) * incy;

    for (int i = 0; i < n; ++i) {
        dy[iy] /= dx[ix];
        ix += incx;
        iy += incy;
    }
}

void runTests() {
    // Test 1
    std::vector<float> dx = {1, 2, 3, 4, 5};
    std::vector<float> dy = {10, 20, 30, 40, 50};
    sedivm(5, dx.data(), 1, dy.data(), 1);
    std::cout << "Test 1: ";
    for (auto& val : dy) std::cout << val << " ";
    std::cout << std::endl;

    // Test 2
    dx = {1, 2, 3, 4, 5};
    dy = {5, 10, 15, 20, 25};
    sedivm(3, dx.data(), 2, dy.data(), 2);
    std::cout << "Test 2: ";
    for (auto& val : dy) std::cout << val << " ";
    std::cout << std::endl;

    // Test 3
    dx = {5, 4, 3, 2, 1};
    dy = {50, 40, 30, 20, 10};
    sedivm(5, dx.data(), -1, dy.data(), -1);
    std::cout << "Test 3: ";
    for (auto& val : dy) std::cout << val << " ";
    std::cout << std::endl;
}

int main() {
    runTests();
    return 0;
}